### ISSUE TITLE
Improving performance of the Library with Cython

### ISSUE THREAD

**Issue by Bhargavasomu** [ethereum/py_ecc#50]
### What is wrong?

There are many places in the code like, `finding an intersection of the tangent with the curve` etc. In other words, there are many functions whose variables make use of **fixed specific datatypes** (mostly `int`). I feel that `Cythonizing` these functions would improve the performance.

### How can it be fixed

Make the suitable functions as `cython functions`.

**Comment by hwwhww** [comment]
Agreed! We did consider using Cython to improve performance; especially that Serenity BLS signature aggregation would be one of the highest overhead.

**Comment by carver** [comment]
> Maybe tackle this after we have benchmarks for the library.

I would go as far as saying this is **blocked** on #49 -- there is no point in starting to optimize performance without a benchmarking suite.
